flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/VectorSearchRuntimeConfigOptions.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api.config;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.configuration.ConfigOption;
+
+import org.apache.flink.shaded.guava33.com.google.common.collect.ImmutableSet;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/**
+ * This class holds option name definitions for VECTOR_SEARCH runtime config based on {@link
+ * ConfigOption}.
+ */
+@PublicEvolving
+public class VectorSearchRuntimeConfigOptions {
+
+    public static final ConfigOption<Boolean> ASYNC =
+            key("async")
+                    .booleanType()
+                    .noDefaultValue()
```

Comment:
Why there's no default value?

Reply:
It's same as MLPredictRuntimeConfigOptions#ASYNC. I think we can align the behaviour.